qt/python/mantidqtinterfaces/mantidqtinterfaces/dns_powder_tof/file_selector/file_selector_presenter.py

```diff
@@ -0,0 +1,281 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+
+"""
+DNS File selector Presenter - Tab of DNS Reduction GUI.
+"""
+
+from mantidqtinterfaces.dns_powder_tof.data_structures.dns_observer import \
+    DNSObserver
+
+
+class DNSFileSelectorPresenter(DNSObserver):
+    def __init__(self,
+                 name=None,
+                 parent=None,
+                 view=None,
+                 model=None,
+                 watcher=None):
+        # pylint: disable=too-many-arguments
+        super().__init__(parent=parent, name=name, view=view, model=model)
+        self.watcher = watcher
+        self.view = view
+        # set sample data view
+        self.view.set_tree_model(self.model.get_model())
+        # set standard data view
+        self.view.set_tree_model(self.model.get_model(standard=True),
+                                 standard=True)
+
+        # adjust view to column width
+        self.num_columns = self.model.get_active_model_column_count()
+        self.view.adjust_treeview_columns_width(self.num_columns)
+
+        self._old_data_set = set()
+
+        # connect signals
+        self.view.sig_expanded.connect(self._expanded)
+
+        self.view.sig_read_all.connect(self._read_all)
+        self.view.sig_filters_clicked.connect(self._filter_scans)
+        self.view.sig_standard_filters_clicked.connect(self._filter_standard)
+        self.view.sig_check_all.connect(self._check_all_visible_scans)
+        self.view.sig_uncheck_all.connect(self._uncheck_all_scans)
+        self.view.sig_check_last.connect(self._check_last_scans)
+        self.view.sig_check_selected.connect(self._check_selected_scans)
+        self.view.sig_right_click.connect(self._right_click)
+        self.view.sig_progress_canceled.connect(self._cancel_loading)
+        self.view.sig_autoload_new_clicked.connect(self._autoload_new)
+        self.view.sig_auto_select_standard_clicked.connect(self._auto_select_standard)
+        self.view.sig_dataset_changed.connect(self._dataset_changed)
+        self.watcher.sig_files_changed.connect(self._files_changed_by_watcher)
+
+    # loading
+    def _read_all(self, filtered=False, watcher=False, start=None, end=None):
+        """
+        Reading of new files, if filtered is True, only the files in the
+        range specified by start and end.
+        """
+        fn_range = [start, end]
```

Comment:
What does fn stand for?

Reply:
It stands for "file number", I'll refactor it